cachi2/core/package_managers/bundler/main.py

```diff
@@ -30,7 +36,11 @@ def fetch_bundler_source(request: Request) -> RequestOutput:
     for package in request.packages:
         path_within_root = request.source_dir.join_within_root(package.path)
         components.extend(
-            _resolve_bundler_package(package_dir=path_within_root, output_dir=request.output_dir)
+            _resolve_bundler_package(
+                package_dir=path_within_root,
+                output_dir=request.output_dir,
+                allow_binary=cast(BundlerPackageInput, package).allow_binary,
```

Comment:
This is unnecessarily difficult to read, I forgot to use `request.bundler_packages` instead of `request.packages` when adding the function. Then `allow_binary=package.allow_binary` should be fine.